py/server/deephaven/table.py

```diff
@@ -503,6 +505,112 @@ def meta_table(self) -> Table:
         return self.j_table
 
+    def iter_dict(self, cols: Optional[Union[str, Sequence[str]]] = None) -> Generator[Dict[str, Any], None, None]:
+        """ Returns a generator that reads one row at a time from the table into a dictionary. The dictionary is a map
+        of column names to scalar values of the column data type.
+
+        If the table is refreshing and no update graph locks are currently being held, the generator will try to acquire
+        the shared lock of the update graph before reading the table data. This provides a consistent view of the data.
+        The side effect of this is that the table will not be able to refresh while the table is being iterated on.
+        Additionally, the generator internally maintains a fill context. The auto acquired shared lock and the fill
+        context will be released after the generator is destroyed. That can happen implicitly when the generator
+        is used in a for-loop. When the generator is not used in a for-loop, to prevent resource leaks, it must be closed
+        after use by either (1) by setting it to None, (2) using the del statement, or (3) calling the close() method on it.
+
+        Args:
+            cols (Optional[Union[str, Sequence[str]]]): The columns to read. If None, all columns are read.
+
+        Returns:
+            A generator that yields a dictionary of column names to scalar values.
+        """
+        from deephaven._table_reader import _table_reader_rows # to prevent circular import
+
+        return _table_reader_rows(self, cols)
+
+    def iter_tuple(self, cols: Optional[Union[str, Sequence[str]]] = None, name: str = 'Deephaven') -> Generator[Tuple[Any], None, None]:
+        """ Returns a generator that reads one row at a time from the table into a named tuple. The named tuple is made
+        up of fields with their names being the column names and their values being of the column data types.
+
+        If the table is refreshing and no update graph locks are currently being held, the generator will try to acquire
+        the shared lock of the update graph before reading the table data. This provides a consistent view of the data.
+        The side effect of this is that the table will not be able to refresh while the table is being iterated on.
+        Additionally, the generator internally maintains a fill context. The auto acquired shared lock and the fill
+        context will be released after the generator is destroyed. That can happen implicitly when the generator
+        is used in a for-loop. When the generator is not used in a for-loop, to prevent resource leaks, it must be closed
+        after use by either (1) by setting it to None, (2) using the del statement, or (3) calling the close() method on it.
+
+        Args:
+            cols (Optional[Union[str, Sequence[str]]]): The columns to read. If None, all columns are read. Default is None.
+            name (str): The name of the named tuple. Default is 'Deephaven'.
```

Comment:
I have some concerns about having a constant default tuple name.  I'm not sure exactly what Python does under the covers.

I asked perplexity:
```
are there any problems if multiple namedtuples in python have the same name?
```

and it spit out:

While it's technically possible to create multiple namedtuples with the same name in Python, doing so can lead to several issues and is generally not recommended. Here are some potential problems:
1. Namespace conflicts: If you create multiple namedtuples with the same name in the same namespace, the later definition will overwrite the earlier one. This can lead to unexpected behavior in your code.
2. Type checking issues: Static type checkers and IDEs may have difficulty distinguishing between different namedtuples with the same name, potentially leading to incorrect type hints and reduced code assistance.
3. Confusion and reduced code readability: Using the same name for different namedtuples can make your code less clear and harder to understand, especially for other developers working on the project.
4. Potential runtime errors: If you're using the namedtuples in different parts of your code, having multiple definitions with the same name could lead to runtime errors if the wrong version is used in a particular context.
5. Serialization and deserialization problems: If you're serializing and deserializing namedtuples, having multiple definitions with the same name could cause issues when trying to reconstruct the objects.

To avoid these problems, it's best to give each namedtuple a unique and descriptive name that reflects its purpose or contents. This practice enhances code clarity, reduces the risk of errors, and improves maintainability.
If you find yourself needing multiple similar namedtuples, consider using inheritance or composition to create related structures, or use a more flexible data structure like a dataclass if appropriate for your use case.

Reply:
I actually thought about it and did some testing. I think what was spit out was probably right in the context of user defined named tuples but not so much in run-time generated ones. Yes, they will have the same class names, but they are separate classes and don't interfere with each other.  And if the users really need to use the class names, they have the option to select unique names.  Pandas uses 'Pandas' for all the named tuples by default too.

Reply:
If you have tested it, and pandas uses a constant, I'm guessing it is fine.